chain/chain/src/signature_verification.rs

```diff
@@ -0,0 +1,48 @@
+use near_chain_primitives::Error;
+use near_crypto::Signature;
+use near_epoch_manager::EpochManagerAdapter;
+use near_primitives::{
+    epoch_block_info::BlockInfo,
+    hash::CryptoHash,
+    sharding::{ChunkHash, ShardChunkHeader},
+    stateless_validation::ChunkProductionKey,
+    types::validator_stake::ValidatorStake,
+};
+use std::sync::Arc;
+
+/// Verify chunk header signature.
+/// return false if the header signature does not match the key for the assigned chunk producer
+/// for this chunk, or if the chunk producer has been slashed
+/// return `EpochError::NotAValidator` if cannot find chunk producer info for this chunk
+pub fn verify_chunk_header_signature(
+    chunk_hash: &ChunkHash,
+    signature: &Signature,
+    chunk_producer: ValidatorStake,
+    block_info: Arc<BlockInfo>,
+) -> Result<bool, Error> {
+    if block_info.slashed().contains_key(chunk_producer.account_id()) {
+        return Ok(false);
+    }
+    Ok(signature.verify(chunk_hash.as_ref(), chunk_producer.public_key()))
+}
+
+pub fn verify_chunk_header_signature_with_epoch_manager(
+    epoch_manager: &dyn EpochManagerAdapter,
+    chunk_header: &ShardChunkHeader,
+    parent_hash: &CryptoHash,
```

Comment:
I think the `parent_hash` here could be replaced by `chunk_header.prev_block_hash()` but I'm not going to complain too much as this is just refactoring.

Reply:
Some tests failed and I also found [this](https://github.com/near/nearcore/blob/bd73c723fca2f98105f8fa595efc20aae15cd4b7/chain/chunks/src/shards_manager_actor.rs#L1353) comment